server/src/com/thoughtworks/go/config/update/CreateSCMConfigCommand.java

```diff
@@ -0,0 +1,79 @@
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.thoughtworks.go.config.update;
+
+import com.thoughtworks.go.config.CruiseConfig;
+import com.thoughtworks.go.config.commands.EntityConfigUpdateCommand;
+import com.thoughtworks.go.domain.scm.SCM;
+import com.thoughtworks.go.domain.scm.SCMs;
+import com.thoughtworks.go.i18n.LocalizedMessage;
+import com.thoughtworks.go.server.domain.Username;
+import com.thoughtworks.go.server.service.GoConfigService;
+import com.thoughtworks.go.server.service.materials.PluggableScmService;
+import com.thoughtworks.go.server.service.result.LocalizedOperationResult;
+import com.thoughtworks.go.serverhealth.HealthStateType;
+
+public class CreateSCMConfigCommand implements EntityConfigUpdateCommand<SCM>{
+
+
+    private final SCM globalScmConfig;
+    private final PluggableScmService pluggableScmService;
+    private SCM preprocessedGlobalScmConfig;
+    private final GoConfigService goConfigService;
+    private final Username currentUser;
+    private final LocalizedOperationResult result;
+
+    public CreateSCMConfigCommand(SCM globalScmConfig, PluggableScmService pluggableScmService, LocalizedOperationResult result, Username currentUser, GoConfigService goConfigService) {
+        this.globalScmConfig = globalScmConfig;
+        this.pluggableScmService = pluggableScmService;
+        this.goConfigService = goConfigService;
+        this.currentUser = currentUser;
+        this.result = result;
+    }
+
+
+    @Override
+    public void update(CruiseConfig modifiedConfig) throws Exception {
+        SCMs scms = modifiedConfig.getSCMs();
+        scms.add(globalScmConfig);
+        modifiedConfig.setSCMs(scms);
+    }
+
+    @Override
+    public boolean isValid(CruiseConfig preprocessedConfig) {
+        SCMs scms = preprocessedConfig.getSCMs();
+        preprocessedGlobalScmConfig = scms.find(globalScmConfig.getSCMId());
+        scms.validate(null);
+        boolean isValid = pluggableScmService.validate(preprocessedGlobalScmConfig);
+
+        return preprocessedGlobalScmConfig.errors().isEmpty() && scms.errors().isEmpty() && isValid;
+    }
+
+    @Override
+    public SCM getPreprocessedEntityConfig() {
+        return preprocessedGlobalScmConfig;
+    }
+
+    @Override
+    public boolean canContinue(CruiseConfig cruiseConfig) {
+         if (!(goConfigService.isUserAdmin(currentUser)) || goConfigService.isGroupAdministrator(currentUser.getUsername())) {
+            result.unauthorized(LocalizedMessage.string("UNAUTHORIZED_TO_EDIT"), HealthStateType.unauthorised());
+            return false;
+        }
+        return true;
```

Comment:
Should we be moving such logic to the `PluggableScmService` instead of here to be consistent with other services?